src/hotspot/share/gc/shenandoah/shenandoahMark.inline.hpp

```diff
@@ -288,4 +293,16 @@ inline void ShenandoahConcurrentMark::mark_through_ref(T *p, ShenandoahHeap* hea
   }
 }
 
-#endif // SHARE_GC_SHENANDOAH_SHENANDOAHCONCURRENTMARK_INLINE_HPP
+ShenandoahObjToScanQueueSet* ShenandoahMark::task_queues() const {
+  return _task_queues;
+}
+
+ShenandoahObjToScanQueue* ShenandoahMark::get_queue(uint index) const {
+  return _task_queues->queue(index);
+}
+
+ShenandoahHeap* ShenandoahMark::heap() const {
+  return _heap;
+}
```

Comment:
Do we really need this method? `ShenandoahHeap::heap()` is supposed to be as fast.

Reply:
No. Quite messy on how to access heap in mark code ... removed _heap member and all access via ShenandoahHeap::heap()